tools/render-test/render-d3d11.cpp

```diff
@@ -561,49 +535,41 @@ InputLayout* D3D11Renderer::createInputLayout(InputElementDesc const* inputEleme
         hlslCursor += sprintf(hlslCursor, "%s a%d : %s%d",
             typeName,
             (int)ii,
-            inputElements[ii].semanticName,
-            (int)inputElements[ii].semanticIndex);
+            inputElementsIn[ii].semanticName,
+            (int)inputElementsIn[ii].semanticIndex);
     }
 
     hlslCursor += sprintf(hlslCursor, "\n) : SV_Position { return 0; }");
 
-    auto dxVertexShaderBlob = compileHLSLShader("inputLayout", hlslBuffer, "main", "vs_5_0");
-    if (!dxVertexShaderBlob)
+    auto vertexShaderBlob = compileHLSLShader("inputLayout", hlslBuffer, "main", "vs_5_0");
+    if (!vertexShaderBlob)
         return nullptr;
 
-    ID3D11InputLayout* dxInputLayout = nullptr;
-    HRESULT hr = dxDevice->CreateInputLayout(
-        &dxInputElements[0],
-        (UINT)inputElementCount,
-        dxVertexShaderBlob->GetBufferPointer(),
-        dxVertexShaderBlob->GetBufferSize(),
-        &dxInputLayout);
+    ID3D11InputLayout* inputLayout = nullptr;
+    HRESULT hr = m_device->CreateInputLayout(&inputElements[0], (UINT)inputElementCount, vertexShaderBlob->GetBufferPointer(), vertexShaderBlob->GetBufferSize(),
+        &inputLayout);
 
-    dxVertexShaderBlob->Release();
+    vertexShaderBlob->Release();
 
     if (FAILED(hr))
         return nullptr;
 
-    return (InputLayout*)dxInputLayout;
+    return (InputLayout*)inputLayout;
 }
 
-void* D3D11Renderer::map(ID3D11Buffer * buffer, MapFlavor flavor)
+void* D3D11Renderer::map(ID3D11Buffer* buffer, MapFlavor flavorIn)
 {
-    auto dxContext = dxImmediateContext;
-
-    auto dxBuffer = buffer;
-
-    D3D11_MAP dxMapFlavor;
-    switch (flavor)
+    D3D11_MAP mapType;
```

Comment:
Another style thing, which actually matters a lot in a compiler codebase: the word "type" can mean a lot of things in different contexts, and within a compiler codebase we usually want to use it with some precision. Many similar words like "kind," "sort," "class," etc. also have precise technical definitions, and so I tend to avoid all of them unless they are being used consistently with those definitions (Slang currently uses "type" for some things that are actually "kinds," but there is at least precedent for that one).

So we are left with the question of what to use when we'd casually say "type," but want to avoid confusion. It turns out that the word "flavor" is maybe the best term around that hasn't already been claimed by the PL community (admittedly there is an object system for Lisp that was called Flavors, and had a concept of "flavors" as something akin to mixin classes).